### DISCUSSION TITLE
How is this different that just installing Docker directly?

### DISCUSSION THREAD

**Discussion by rowanmoul** [Jip-Hop/jailmaker#63] [General]
As I understand it `systemd-nspawn` is more or less just another container runtime, not so different from containerd, LXC, etc.
Since Debian packages it separately from their main `systemd` package, using Jailmaker requires installing new packages on TrueNAS Scale (a big no-no as far as ix-systems is concerned). If you're going off-script anyway, why not just install docker directly and be done with it?

What is the benefit of docker-in-`systemd-nspawn` vs just docker?

**Comment by Jip-Hop** [comment]
With jailmaker you can run docker but you don't have to, you can install anything you like inside. So it has this flexibility.

Besides, last time I checked systemd-nspawn was (again) included with SCALE so you don't have to install anything which isn't already there.

If it some point the systemd-container package isn't installed with SCALE by default the jailmaker script will install it, this is true. But compared to installing docker it has way less dependencies (none which aren't already installed if I'm correct) compared to installing docker.

So by using systemd-nspawn you need to add nothing (or very little) compared to installing docker and the risk of breaking the SCALE system is much smaller.

**Reply by rowanmoul** [reply]
You are correct, it is actually installed already! I'm so used to having to dig for basic tools like `smartctl` or `zfs`, that I managed to forget to just try to tab-complete it, or use `which`...
Time to read through `jlmkr.py` in much more detail, as well as read up on more complex networking scenarios (I like to understand what I am doing, not just blindly treat scripts like yours as a black box)
